applications/MappingApplication/custom_utilities/interface_condition.h

```diff
@@ -93,89 +89,206 @@ namespace Kratos
+                  && mNumPoints == 2) { // I am a linear line condition
+                  is_inside = MapperUtilities::ProjectPointToLine(mpCondition, rGlobalCoords,
                                                                   projection_local_coords,
-                                                                  projection_distance, m_normal);
-          } else if (m_geometry_family == GeometryData::Kratos_Quadrilateral) { // I am a quadrilateral condition
-              is_inside = MapperUtilities::ProjectPointToQuaddrilateral(m_p_condition, global_coords,
-                                                                        projection_local_coords,
-                                                                        projection_distance, m_normal);
-          } else {
-              KRATOS_ERROR << "MappingApplication; InterfaceCondition; \"EvaluateResult\" Unsupported Geometry!" << std::endl;
+                                                                  projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Triangle 
+                        && mNumPoints == 3) { // I am a linear triangular condition
+                  is_inside = MapperUtilities::ProjectPointToTriangle(mpCondition, rGlobalCoords,
+                                                                      projection_local_coords,
+                                                                      projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Quadrilateral
+                        && mNumPoints == 4) { // I am a linear quadrilateral condition
+                  is_inside = MapperUtilities::ProjectPointToQuadrilateral(mpCondition, rGlobalCoords,
+                                                                          projection_local_coords,
+                                                                          projection_distance);
+              } else if (mGeometryFamily == GeometryData::Kratos_Tetrahedra ||
+                         mGeometryFamily == GeometryData::Kratos_Prism ||
+                         mGeometryFamily == GeometryData::Kratos_Hexahedra) { // Volume Mapping
+                  is_inside = MapperUtilities::PointLocalCoordinatesInVolume(mpCondition, rGlobalCoords,
+                                                                             projection_local_coords,
+                                                                             projection_distance);
+              } else {
+                  std::cout << "MAPPER WARNING, Unsupported geometry, " 
+                            << "using an approximation" << std::endl;
+                  return false;
+              }
+          } catch (...) {
+              std::cout << "MAPPER WARNING, Functions necessary for projection "
+                        << "are not implemented in the core, " 
+                        << "using an approximation" << std::endl;
+              return false;
           }
```

Comment:
I want to have the mapper working in every case.
This particular catch is intended in case a projection can not be done because there are some functions missing in the core, which might be the case for not so commonly used geometries.
In this case I dont want the mapper to be the limiting factor, it should still work, with only approximations though